Add labels to docker build commands

nemchik opened the pull request:
Ref: https://github.com/opencontainers/image-spec/blob/master/annotations.md#pre-defined-annotation-keys

Roxedus reviewed (approved):
LGTM, follows OCI spec

nemchik commented:
Rebased and changed target to the monthly deprecations branch. I will merge this into https://github.com/linuxserver/docker-jenkins-builder/pull/101 which can then merge into https://github.com/linuxserver/docker-jenkins-builder/pull/98
